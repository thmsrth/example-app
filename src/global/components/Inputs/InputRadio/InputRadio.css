.InputRadio__container {
  display: flex;
}

.FormInput__label.InputRadioItem__label {
  display: flex;
  padding-top: 0;
  align-items: baseline;
}

.FormInput__input.InputRadioItem__input {
  flex: none;
  width: auto;
}
