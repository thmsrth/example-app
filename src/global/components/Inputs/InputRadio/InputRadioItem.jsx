import React from 'react';
import PropTypes from 'prop-types';

import { bemCls } from '../../../utils/ClassNameHelpers';

const InputRadioItem = ({
  className, name, label, value, checked, onChange, onBlur,
}) => (
  <div className={`InputRadioItem ${className}`}>
    <label
      className={`${bemCls(`InputRadioItem ${className}`, '__label')} ${
        bemCls(`InputRadioItem ${className}`, `__${name}__label`)}`}
    >
      <input
        type="radio"
        className={`${bemCls(`InputRadioItem ${className}`, '__input')} ${
          bemCls(`InputRadioItem ${className}`, `__${name}__input`)}`}
        name={name}
        value={value}
        checked={checked}
        onChange={onChange}
        onBlur={onBlur}
      />
      {label}
    </label>
  </div>
);

export default InputRadioItem;

InputRadioItem.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  className: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]),
  checked: PropTypes.bool,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

InputRadioItem.defaultProps = {
  label: '',
  className: '',
  value: '',
  checked: false,
  onChange: () => {},
  onBlur: () => {},
};
