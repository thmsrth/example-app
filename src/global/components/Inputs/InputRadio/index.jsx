import React from 'react';

import InputRadio from './InputRadio';

const InputRadioWrapper = props => (
  <InputRadio {...props} />
);

export default InputRadioWrapper;
