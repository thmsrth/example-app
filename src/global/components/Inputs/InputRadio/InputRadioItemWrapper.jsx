import React from 'react';

import PropTypes from 'prop-types';
import validationKeysType from '../../Form/types/validationKeys';
import validationKeysDefaults from '../../Form/defaults/validationKeys';

import InputRadioItem from './InputRadioItem';

class InputRadioItemWrapper extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
  }

  handleChange(event) {
    const { name, option, onChange } = this.props;
    if (event.target.checked) onChange(name, option.value);
  }

  handleBlur() {
    const {
      name, validationKeys, onBlur,
    } = this.props;
    const options = { type: 'radio' };

    onBlur(name, validationKeys, options);
  }

  render() {
    const { value, option } = this.props;
    return (
      <InputRadioItem
        {...this.props}
        value={option.value}
        checked={value === option.value}
        label={option.label}
        onChange={this.handleChange}
        onBlur={this.handleBlur}
      />
    );
  }
}

export default InputRadioItemWrapper;

InputRadioItemWrapper.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.bool,
  ]),
  name: PropTypes.string,
  validationKeys: validationKeysType,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  option: PropTypes.shape({
    value: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string,
      PropTypes.bool,
    ]),
    label: PropTypes.string,
  }),
};

InputRadioItemWrapper.defaultProps = {
  value: null,
  name: '',
  validationKeys: validationKeysDefaults,
  option: null,
  onChange: () => {},
  onBlur: () => {},
};
