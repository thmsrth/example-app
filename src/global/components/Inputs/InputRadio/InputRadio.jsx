import React from 'react';

import PropTypes from 'prop-types';
import validationKeysType from '../../Form/types/validationKeys';
import validationKeysDefaults from '../../Form/defaults/validationKeys';

import { bemCls } from '../../../utils/ClassNameHelpers';
import InputRadioItemWrapper from './InputRadioItemWrapper';

import './InputRadio.css';

const InputRadio = ({
  label, options, name, className, validationKeys, ...otherProps
}) => {
  const {
    required,
    validationAlert,
    warningAlert,
  } = validationKeys || InputRadio.defaultProps.validationKeys;

  return (
    <div className={`InputRadio ${className} ${
      bemCls(`InputRadio ${className}`, `__${name}`)} ${
      bemCls(`InputRadio ${className}`, '__required', required)} ${
      bemCls(`InputRadio ${className}`, '--alert', validationAlert)}`}
    >
      <div className={bemCls(`InputRadio ${className}`, '__label')}>
        {`${label}${required ? '*' : ''}`}
      </div>
      <div className={`${bemCls(`InputRadio ${className}`, '__alertMsg')} ${
        bemCls(`InputRadio ${className}`, `__${name}__alertMsg`)}`}
      >
        {validationAlert}
      </div>
      <div className={bemCls(`InputRadio ${className}`, '__container')}>
        {options.map(option => (
          <InputRadioItemWrapper
            {...otherProps}
            className={className}
            name={name}
            key={option.value}
            option={option}
            validationKeys={validationKeys}
          />
        ))}
      </div>
      <div className={`${bemCls(`InputRadio ${className}`, '__warningMsg')} ${
        bemCls(`InputRadio ${className}`, `__${name}__warningMsg`)}`}
      >
        {warningAlert}
      </div>
    </div>
  );
};

export default InputRadio;

InputRadio.propTypes = {
  label: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
      PropTypes.bool,
    ]),
    label: PropTypes.string,
  })),
  className: PropTypes.string,
  validationKeys: validationKeysType,
  name: PropTypes.string,
};

InputRadio.defaultProps = {
  label: '',
  className: '',
  options: [],
  validationKeys: validationKeysDefaults,
  name: '',
};
