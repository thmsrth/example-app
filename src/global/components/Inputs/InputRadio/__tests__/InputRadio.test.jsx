import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputRadio from '../InputRadio';

const props = {
  className: 'MyInputRadio',
  name: 'type',
  label: 'My Input Label',
  validationKeys: {
    required: false,
    validationAlert: '',
    warningAlert: '',
  },
  onChange: () => {},
  options: [
    {
      value: 'EMPLOYEE',
      label: 'EMPLOYEE',
    },
    {
      value: 'MEMBER',
      label: 'MEMBER',
    },
    {
      value: 'GUEST',
      label: 'GUEST',
    },
  ],
  value: 'EMPLOYEE',
};

describe('InputRadio', () => {
  it('should render radio items components for each option correctly', () => {
    const tree = shallow(<InputRadio {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component as a required input not checked', () => {
    const tree = shallow(<InputRadio
      {...props}
      validationKeys={{ required: true }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a validation alert message', () => {
    const tree = shallow(<InputRadio
      {...props}
      validationKeys={{ validationAlert: 'My validation alert message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a warning message', () => {
    const tree = shallow(<InputRadio
      {...props}
      validationKeys={{ warningAlert: 'My warning message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
