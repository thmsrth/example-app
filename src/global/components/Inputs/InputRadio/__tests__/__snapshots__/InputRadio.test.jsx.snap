// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRadio should render radio items components for each option correctly 1`] = `
<div
  className="InputRadio MyInputRadio InputRadio__type MyInputRadio__type  "
>
  <div
    className="InputRadio__label MyInputRadio__label"
  >
    My Input Label
  </div>
  <div
    className="InputRadio__alertMsg MyInputRadio__alertMsg InputRadio__type__alertMsg MyInputRadio__type__alertMsg"
  />
  <div
    className="InputRadio__container MyInputRadio__container"
  >
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="EMPLOYEE"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "EMPLOYEE",
          "value": "EMPLOYEE",
        }
      }
      validationKeys={
        Object {
          "required": false,
          "validationAlert": "",
          "warningAlert": "",
        }
      }
      value="EMPLOYEE"
    />
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="MEMBER"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "MEMBER",
          "value": "MEMBER",
        }
      }
      validationKeys={
        Object {
          "required": false,
          "validationAlert": "",
          "warningAlert": "",
        }
      }
      value="EMPLOYEE"
    />
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="GUEST"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "GUEST",
          "value": "GUEST",
        }
      }
      validationKeys={
        Object {
          "required": false,
          "validationAlert": "",
          "warningAlert": "",
        }
      }
      value="EMPLOYEE"
    />
  </div>
  <div
    className="InputRadio__warningMsg MyInputRadio__warningMsg InputRadio__type__warningMsg MyInputRadio__type__warningMsg"
  />
</div>
`;

exports[`InputRadio should render the component as a required input not checked 1`] = `
<div
  className="InputRadio MyInputRadio InputRadio__type MyInputRadio__type InputRadio__required MyInputRadio__required InputRadio--alert MyInputRadio--alert"
>
  <div
    className="InputRadio__label MyInputRadio__label"
  >
    My Input Label*
  </div>
  <div
    className="InputRadio__alertMsg MyInputRadio__alertMsg InputRadio__type__alertMsg MyInputRadio__type__alertMsg"
  />
  <div
    className="InputRadio__container MyInputRadio__container"
  >
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="EMPLOYEE"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "EMPLOYEE",
          "value": "EMPLOYEE",
        }
      }
      validationKeys={
        Object {
          "required": true,
        }
      }
      value="EMPLOYEE"
    />
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="MEMBER"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "MEMBER",
          "value": "MEMBER",
        }
      }
      validationKeys={
        Object {
          "required": true,
        }
      }
      value="EMPLOYEE"
    />
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="GUEST"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "GUEST",
          "value": "GUEST",
        }
      }
      validationKeys={
        Object {
          "required": true,
        }
      }
      value="EMPLOYEE"
    />
  </div>
  <div
    className="InputRadio__warningMsg MyInputRadio__warningMsg InputRadio__type__warningMsg MyInputRadio__type__warningMsg"
  />
</div>
`;

exports[`InputRadio should render the component with a validation alert message 1`] = `
<div
  className="InputRadio MyInputRadio InputRadio__type MyInputRadio__type InputRadio__required MyInputRadio__required InputRadio--alert MyInputRadio--alert"
>
  <div
    className="InputRadio__label MyInputRadio__label"
  >
    My Input Label
  </div>
  <div
    className="InputRadio__alertMsg MyInputRadio__alertMsg InputRadio__type__alertMsg MyInputRadio__type__alertMsg"
  >
    My validation alert message
  </div>
  <div
    className="InputRadio__container MyInputRadio__container"
  >
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="EMPLOYEE"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "EMPLOYEE",
          "value": "EMPLOYEE",
        }
      }
      validationKeys={
        Object {
          "validationAlert": "My validation alert message",
        }
      }
      value="EMPLOYEE"
    />
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="MEMBER"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "MEMBER",
          "value": "MEMBER",
        }
      }
      validationKeys={
        Object {
          "validationAlert": "My validation alert message",
        }
      }
      value="EMPLOYEE"
    />
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="GUEST"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "GUEST",
          "value": "GUEST",
        }
      }
      validationKeys={
        Object {
          "validationAlert": "My validation alert message",
        }
      }
      value="EMPLOYEE"
    />
  </div>
  <div
    className="InputRadio__warningMsg MyInputRadio__warningMsg InputRadio__type__warningMsg MyInputRadio__type__warningMsg"
  />
</div>
`;

exports[`InputRadio should render the component with a warning message 1`] = `
<div
  className="InputRadio MyInputRadio InputRadio__type MyInputRadio__type InputRadio__required MyInputRadio__required InputRadio--alert MyInputRadio--alert"
>
  <div
    className="InputRadio__label MyInputRadio__label"
  >
    My Input Label
  </div>
  <div
    className="InputRadio__alertMsg MyInputRadio__alertMsg InputRadio__type__alertMsg MyInputRadio__type__alertMsg"
  />
  <div
    className="InputRadio__container MyInputRadio__container"
  >
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="EMPLOYEE"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "EMPLOYEE",
          "value": "EMPLOYEE",
        }
      }
      validationKeys={
        Object {
          "warningAlert": "My warning message",
        }
      }
      value="EMPLOYEE"
    />
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="MEMBER"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "MEMBER",
          "value": "MEMBER",
        }
      }
      validationKeys={
        Object {
          "warningAlert": "My warning message",
        }
      }
      value="EMPLOYEE"
    />
    <InputRadioItemWrapper
      className="MyInputRadio"
      key="GUEST"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      option={
        Object {
          "label": "GUEST",
          "value": "GUEST",
        }
      }
      validationKeys={
        Object {
          "warningAlert": "My warning message",
        }
      }
      value="EMPLOYEE"
    />
  </div>
  <div
    className="InputRadio__warningMsg MyInputRadio__warningMsg InputRadio__type__warningMsg MyInputRadio__type__warningMsg"
  >
    My warning message
  </div>
</div>
`;
